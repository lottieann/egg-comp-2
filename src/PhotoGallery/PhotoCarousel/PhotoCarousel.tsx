import * as React from 'react'

import styled from 'styled-components'
import Image from 'next/image'

export interface PhotoCarouselProps {
  photoInfo: any
}

const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin-top: 20vh;

  p {
    font-family: Helvetica;
    font-size: 18px;
    text-align: center;
  }
`

const StyledImage = styled(Image)`
  height: 300px;
  width: 300px;
  padding: 10px;
`
const PhotoContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const NextButton = styled.button`
  width: 0;
  height: 0;
  border-top: 30px solid transparent;
  border-bottom: 30px solid transparent;
  border-left: 30px solid grey;
  border-right: transparent;
  background-color: transparent;
  margin-left: 10px;
`
const PreviousButton = styled.button`
  width: 0;
  height: 0;
  border-top: 30px solid transparent;
  border-bottom: 30px solid transparent;
  border-right: 30px solid grey;
  border-left: transparent;
  background-color: transparent;
  margin-right: 10px;
`

const ButtonSpace = styled.div`
  height: 30px;
  width: 30px;
  background-color: transparent;
`

const EntryPhoto = styled.div`
  height: 300px;
  width: 300px;
  background-color: lightgray;
  border: 2px solid gray;
  padding: 10px;
`

export const PhotoCarousel = () => {
  const [photoIndex, setPhotoIndex] = React.useState(0)
  const [isError, setIsError] = React.useState(false)


//const images = [`${props.photoInfo.photo1}`, `${props.photoInfo.photo2}`, `${props.photoInfo.photo3}`]

const images = ['/tigerKing.jpg', '/feathers.jpg', '/plane.jpg', '/beautySalon.jpg', '/grenade.jpg', '/eggcalibur.jpg', '/boot.jpg']

  return (
    <Container>
      <PhotoContainer>
        {photoIndex <= 0 ? (
          <ButtonSpace data-testid="previous-button-space" />
        ) : (
          <PreviousButton
            data-testid="previous-button"
            onClick={() => setPhotoIndex(photoIndex - 1)}
          />
        )}
        {isError ? (
          <EntryPhoto />
        ) : (
          <StyledImage
            data-testid={`image-${photoIndex}`}
            src={`${images[photoIndex]}`}
            alt={`egg-${photoIndex}`}
            width={300}
            height={300}
            onError={() => setIsError(true)}
          />
        )}
        {photoIndex >= 6 ? (
          <ButtonSpace data-testid="next-button-space" />
        ) : (
          <NextButton
            data-testid="next-button"
            onClick={() => setPhotoIndex(photoIndex + 1)}
          />
        )}
      </PhotoContainer>
      <p>Please enjoy some photos from previous years. Return here Easter Saturday to see 2021 hopefuls!</p>
    </Container>
  )
}
