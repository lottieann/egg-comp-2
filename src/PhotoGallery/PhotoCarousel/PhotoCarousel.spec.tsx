import * as React from 'react'

import 'jest-styled-components'

import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { PhotoCarousel, PhotoCarouselProps } from './PhotoCarousel'

afterEach(cleanup)

const photoInfo = {
  entryName: 'Test Entry',
  entryNumber: 0,
  photo1: '/0.jpg',
  photo2: '/1.jpg',
  photo3: '/3.jpg',
  vote1: '',
  vote2: '',
  vote4: '',
  vote3: '',
  vote5: '',
}

const setup = (props: PhotoCarouselProps) => {
  return <PhotoCarousel {...props}/>
}

describe('PhotoCarousel', () => {
  it('should increase photo index when next button is clicked', () => {
    const { getByTestId } = render(setup({
      photoInfo
    }))
    expect(getByTestId('image-0')).toBeInTheDocument()
    fireEvent.click(getByTestId('next-button'))
    expect(getByTestId('image-1')).toBeInTheDocument()
  })
  it('should not increase photo index when maximum is reached', () => {
    const { getByTestId } = render(setup({
      photoInfo
    }))
    expect(getByTestId('image-0')).toBeInTheDocument()
    fireEvent.click(getByTestId('next-button'))
    fireEvent.click(getByTestId('next-button'))
 
    expect(getByTestId("next-button-space")).toBeInTheDocument()
    expect(getByTestId('image-2')).toBeInTheDocument()
  })
  it.skip('should decrease photo index when previous button is clicked', () => {
    const { getByTestId } = render(setup({
      photoInfo
    }))
    expect(getByTestId('image-1')).toBeInTheDocument()
    fireEvent.click(getByTestId('previous-button'))
    expect(getByTestId('image-0')).toBeInTheDocument()
  })
  it.skip('should not decrease photo index when minimum is reached', () => {
    const { getByTestId } = render(setup({
      photoInfo
    }))
    expect(getByTestId('image-1')).toBeInTheDocument()
    fireEvent.click(getByTestId('previous-button'))
 
    expect(getByTestId("previous-button-space")).toBeInTheDocument()
    expect(getByTestId('image-0')).toBeInTheDocument()
  })
  it('should render PhotoCarousel correctly', () => {
    const { container } = render(setup({
      photoInfo
    }))
    expect(container).toMatchSnapshot()
  })
})
