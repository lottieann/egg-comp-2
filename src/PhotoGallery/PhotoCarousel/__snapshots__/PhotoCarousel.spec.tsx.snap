// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCarousel should render PhotoCarousel correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 20vh;
}

.c3 {
  height: 300px;
  width: 300px;
  padding: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  width: 0;
  height: 0;
  border-top: 30px solid transparent;
  border-bottom: 30px solid transparent;
  border-left: 30px solid grey;
  border-right: transparent;
  background-color: transparent;
  margin-left: 10px;
}

.c2 {
  height: 30px;
  width: 30px;
  background-color: transparent;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="previous-button-space"
      />
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="egg-0"
          class="c3"
          data-testid="image-0"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
      <button
        class="c4"
        data-testid="next-button"
      />
    </div>
  </div>
</div>
`;
